pydantic/_internal/_model_construction.py

```diff
@@ -216,7 +216,7 @@ def __getattr__(self, item: str) -> Any:
             raise AttributeError(item)
 
     @classmethod
-    def __prepare__(cls, *args: Any, **kwargs: Any) -> Mapping[str, object]:
+    def __prepare__(cls, *args: Any, **kwargs: Any) -> MutableMapping[str, object]:
```

Comment:
```suggestion
    def __prepare__(cls, *args: Any, **kwargs: Any) -> _ModelNamespaceDict:
```
And make `_ModelNamespaceDict` parametrized? Could be `_ModelNamespaceDict(dict[str, object])` but that doesn't play nice with Python  < 3.10, so maybe something like `_ModelNamespaceDict(dict, typing.MutableMapping[str, object])`?

Reply:
```python
from typing import MutableMapping

class D(dict, MutableMapping[str, float]):
    pass

d = D()
d["k"] = 100
d[2] = 20
```
does not seem to work, `mypy` finds no issue with that, while it complains as expected for
```python
class D(dict[str, float]):
    pass

d = D()
d["k"] = 100
d[2] = 20
```
`dict[str, object]` works starting with Python 3.9, but not 3.8, which is still supported by `pydantic`.

Reply:
Sorry, I'm not too familiar with the code base, so I only noticed this now: `_model_construction.py` already uses [PEP 585](https://peps.python.org/pep-0585/) type annotations. For consistency, I assume we can annotate the return type as `dict[str, object]` without using `typing.Dict`.